core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -3270,7 +3270,19 @@ compare_address(const struct addr_pool *addr_pool_entry, const char *addr)
         return false;
     }
 
-    uint32 mask = addr_pool_entry->mask;
+    const uint32 max_mask_value = 32;
+    /* no support for invalid mask values */
+    if (addr_pool_entry->mask > max_mask_value) {
+        return false;
+    }
+
+    /* convert mask number into 32-bit mask value, i.e. mask /24 will be
+    converted to 4294967040 (binary: 11111111 11111111 11111111 00000000) */
+    uint32 mask = 0;
+    for (int i = 0; i < addr_pool_entry->mask; i++) {
+        mask |= 1 << (max_mask_value - 1 - i);
```

Comment:
It doesn't feel right. `mask` is used to identify the subnetwork. `first_address` is the first value in the subnetwork which should end with a 0, `last_address` is the last value in the range.

For example:
/0   --> 0xFFFF FFFF 
/1   --> 0xFFFF FFFE
/2   --> 0xFFFF FFFC
/24 --> 0xFF00 0000 
/31 --> 0x8000 0000
/32 --> 0x0, actually it is used for broadcast

Reply:
you are right. `mask` should be the amount number of `1` of the subnet address.